src/hash_map.c

```diff
@@ -255,6 +255,13 @@ rcutils_hash_map_init(
     return RCUTILS_RET_INVALID_ARGUMENT;
   }
 
+  // Due to an optimization we use during lookup, we can currently only handle power-of-two
+  // capacities.  Enforce that here.
+  if ((initial_capacity % 2) != 0) {
```

Comment:
Is it to handle `multiple-of-two` or `power-of-two`?
I think `initial_capacity % 2` can only handle the former. If so, maybe we can use `initial_capacity & ~(SIZE_MAX - 1)` instead.

Reply:
Very good point.  I'll fix it like you suggest.

Reply:
That's a very good point, I totally didn't think about this properly.

That said, I don't think your solution is correct.  If I run it locally, it still returns `true` for other multiple-of-two, like 6.  What does seem to work properly is the following:

```
if ((initial_capacity & (initial_capacity - 1)) == 0) {
```

I'm going to change the code to that.

Reply:
And see 404836c now.